rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+
+import gevent
+import requests
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import AssetWithOracles
+from rotkehlchen.assets.converters import asset_from_bybit
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.constants.timing import DAY_IN_SECONDS, WEEK_IN_SECONDS
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.db.settings import CachedSettings
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.exchanges.data_structures import AssetMovement, MarginPosition, Trade
+from rotkehlchen.exchanges.exchange import ExchangeInterface, ExchangeQueryBalances
+from rotkehlchen.exchanges.utils import pair_symbol_to_base_quote
+from rotkehlchen.history.deserialization import deserialize_price
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_fee,
+    deserialize_fval,
+)
+from rotkehlchen.types import (
+    ApiKey,
+    ApiSecret,
+    AssetMovementCategory,
+    ExchangeAuthCredentials,
+    Fee,
+    Location,
+    Timestamp,
+    TimestampMS,
+    TradeType,
+)
+from rotkehlchen.utils.misc import ts_ms_to_sec, ts_now, ts_now_in_ms, ts_sec_to_ms
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.history.events.structures.base import HistoryEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+PAGINATION_LIMIT: Final = 48
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def bybit_symbol_to_base_quote(symbol: str) -> tuple[AssetWithOracles, AssetWithOracles]:
+    """Turns a gemini symbol product into a base/quote asset tuple
+
+    - Can raise UnprocessableTradePair if symbol is in unexpected format
+    - Case raise UnknownAsset if any of the pair assets are not known to rotki
+    """
+    five_letter_assets = ('SUSHI', '1INCH', 'MATIC')
+    base_asset, quote_asset = pair_symbol_to_base_quote(
+        symbol=symbol,
+        asset_deserialize_fn=asset_from_bybit,
+        five_letter_assets=five_letter_assets,
+    )
+
+    return base_asset, quote_asset
+
+
+class Bybit(ExchangeInterface):
+    def __init__(
+            self,
+            name: str,
+            api_key: ApiKey,
+            secret: ApiSecret,
+            database: 'DBHandler',
+            msg_aggregator: 'MessagesAggregator',
+    ):
+        super().__init__(
+            name=name,
+            location=Location.BYBIT,
+            api_key=api_key,
+            secret=secret,
+            database=database,
+        )
+        self.uri = 'https://api.bybit.com/v5'
+        self.msg_aggregator = msg_aggregator
+        self.session.headers.update({'Content-Type': 'application/json', 'X-BAPI-SIGN-TYPE': '2'})
+        self.recv_window = str(5000)
```

Comment:
What is this? Add a docstring to the constructor explaining the recv_window, why the `X-BAPI-SIGN-TYPE` header is needed, whatis `is_unified_account` what they do and a link to the docs.

Also if the `recv_window` is a constant why not just make it a module constant that is final?

Reply:
`is_unified_account` is needed because the parametres used to query the endpoints are different when the api keys are for a unified account or not. 

The recv windows is the max difference allowed for the timestamps used in the signature and the timne in the server. 

Will link to docs